tests/util/test_data_catalog_helper.py

```diff
@@ -187,3 +187,31 @@ def test_split_missing_main_cm_data():
 
     assert main_cm_data == main_cm_data_expected
     assert prod_cm_data == prod_cm_data_expected
+
+
+def test_format_product_cm_name_sanity():
+    """Unit test case for product name formatting"""
+    product_name = "dummy-valid-1"
+    config_map = "cm"
+    assert format_product_cm_name(config_map, product_name) == f"{config_map}-{product_name}"
+
+
+def test_format_product_name_transform():
+    """Unit test case for valid product name transformation"""
+    product_name = "23dummy_valid-1.x86"
+    config_map = "cm"
+    assert format_product_cm_name(config_map, product_name) == f"{config_map}-23dummy-valid-1.x86"
+
+
+def test_format_product_name_invalid_cases():
+    """Unit test case for valid product name transformation"""
+
+    # case with special characters
+    product_name = "po90-$_invalid"
+    config_map = "cm"
+    assert format_product_cm_name(config_map, product_name) == ""
+
+    # large name cases
+    product_name = "ola-9" * 60
+    config_map = "cm"
+    assert format_product_cm_name(config_map, product_name) == ""
```

Comment:
Can there be any case where `config_map` passed is empty/None ?

Reply:
Seems like they are sending a default value in case its not set:
```
CONFIG_MAP = os.environ.get("CONFIG_MAP", "cray-product-catalog").strip()
PRODUCT_CONFIG_MAP = format_product_cm_name(CONFIG_MAP, PRODUCT)
```
Any way adding a check for empty string, for None logic can fail.
@krabhi-hpe